PathPlanning/DStarLite/DStarLite.py

```diff
@@ -0,0 +1,278 @@
+"""
+D* Lite grid planning
+author: vss2sn (28676655+vss2sn@users.noreply.github.com)
+Link to papers:
+D* Lite (Link: http://idm-lab.org/bib/abstracts/papers/aaai02b.pd)
+Improved Fast Replanning for Robot Navigation in Unknown Terrain
+(Link: http://www.cs.cmu.edu/~maxim/files/dlite_icra02.pdf)
+Implemented maintaining similarity with the pseudocode for understanding
+"""
+from __future__ import annotations
+import math
+import matplotlib.pyplot as plt
+import random
+
+show_animation = True
+pause_time = 0.1
+p_create_random_obstacle = 0
+
+
+class Node:
+    def __init__(self, x: int = 0, y: int = 0, cost: float = 0.0):
+        self.x = x
+        self.y = y
+        self.cost = cost
+
+
+def add_coordinates(node1: Node, node2: Node):
+    new_node = Node()
+    new_node.x = node1.x + node2.x
+    new_node.y = node1.y + node2.y
+    new_node.cost = node1.cost + node2.cost
+    return new_node
+
+
+def compare_coordinates(node1: Node, node2: Node):
+    return node1.x == node2.x and node1.y == node2.y
+
+
+class DStarLite:
+
+    motions = [
+        Node(1, 0, 1),
+        Node(0, 1, 1),
+        Node(-1, 0, 1),
+        Node(0, -1, 1)
+    ]
+
+    def __init__(self, x_max: int, y_max: int, obstacles: list,
+                 spoofed_obstacles: list = list()):
```

Comment:
list should not used for an argument: Ref: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
Will modify.